pkg/kubelet/apis/config/types.go

```diff
@@ -74,6 +74,8 @@ const (
 type KubeletConfiguration struct {
 	metav1.TypeMeta
 
+	// enableServer enables the Kubelet's server
+	EnableServer bool
```

Comment:
do we actually need this in the config or is it implied if RunOnce is false?

Reply:
hmm... it looks like there are three things the kubelet serves:
* secured port, governed by EnableServer
* insecure port, governed by ReadOnlyPort
* pod resources, governed by KubeletPodResources

those have evolved over time and aren't a very coherent config surface... I was hoping this would be improved when it moved into config rather than just replicated.

Reply:
What did you have in mind?

Our main goal right now is to get to a versioned config format. Some refactoring is ok, but I'm wary of blocking on refactorings where the right thing to do may not be well-defined.

Reply:
I don't see a simplistic way to clean that up. I guess if v1beta1 is roughly equivalent to flags, and cleanup happens in later config versions, that could be ok